mailbox/plugin/deleted-messages-vault/src/main/java/org/apache/james/vault/blob/BlobStoreDeletedMessageVault.java

```diff
@@ -79,6 +79,7 @@ public BlobStoreDeletedMessageVault(DeletedMessageMetadataVault messageMetadataV
         return Flux.from(messageMetadataVault.listRelatedBuckets())
             .concatMap(bucketName -> Flux.from(messageMetadataVault.listMessages(bucketName, user)))
             .map(DeletedMessageWithStorageInformation::getDeletedMessage)
+            .filter(deletedMessage -> user.equals(deletedMessage.getOwner()))
```

Comment:
When does `Flux.from(messageMetadataVault.listMessages(bucketName, user)` returns things not related to the user?

I'd love to have a test explaining me that change...

Reply:
yeah seem like I didn't read the code carefully